src/BlockExtractor.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace WP_CLI\I18n;
+
+use Gettext\Extractors\Extractor;
+use Gettext\Extractors\ExtractorInterface;
+use Gettext\Translations;
+use WP_CLI;
+
+final class BlockExtractor extends Extractor implements ExtractorInterface {
+	use IterableCodeExtractor;
+
+	/**
+	 * @inheritdoc
+	 */
+	public static function fromString( $string, Translations $translations, array $options = [] ) {
+		$file = $options['file'];
+		WP_CLI::debug( "Parsing file {$file}" );
+
+		$file_data = json_decode( $string, true );
+
+		if ( null === $file_data ) {
+			WP_CLI::debug(
+				sprintf(
+					'Could not parse file %1$s: error code %2$s',
+					$file,
+					json_last_error()
+				)
+			);
+
+			return;
+		}
+
+		$domain = isset( $file_data['textDomain'] ) ? $file_data['textDomain'] : null;
+
+		// Allow missing domain, but skip if they don't match.
+		if ( null !== $domain && $domain !== $translations->getDomain() ) {
+			return;
+		}
+
+		foreach ( $file_data as $key => $original ) {
+			switch ( $key ) {
+				case 'title':
+				case 'description':
+					$translation = $translations->insert( sprintf( 'block %s', $key ), $original );
+					$translation->addReference( $file );
+					break;
+				case 'keywords':
+					if ( ! is_array( $original ) ) {
+						continue 2;
+					}
+
+					foreach ( $original as $msg ) {
+						$translation = $translations->insert( 'block keyword', $msg );
+						$translation->addReference( $file );
+					}
+
+					break;
+				case 'styleVariations':
```

Comment:
Should check for `styles` too...? Per https://github.com/wp-cli/i18n-command/issues/163#issuecomment-571968841

cc @gziolo

Reply:
Good catch, let's support both to align with https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#style-variations

> Property: `styles`
> Alias: `styleVariations`

Reply:
Done in 2507e8db3459eb6fb3cd6e88b6ff7b8c3386b8b1